2.18 .packages update (#4198)
dart-lang/site-www: src/tools/pub/cmd/pub-get.md

in immediate dependencies (`--precompile`).
To prevent precompilation, use `--no-precompile`.

### `--legacy-packages-file`

> If you utilize any third-party tools with a legacy dependency on `.packages` (desupported in [2.18](/guides/whats-new)),
> use `--legacy-packages-file` to generate a `.packages` file.

Reviewer: I would rather not link to the "What's new" page as that is more about what is new on the website, not the Dart SDK itself. Support is being removed in Dart 2.19 so perhaps we can just include the version this is useful for as a main part of the sentence?

Perhaps something like:

```suggestion
If you are using Dart 2.18 and have any third-party tools
which still use the discontinued `.packages` file,
use `--legacy-packages-file` to generate a `.packages` file.
```

> This support will be removed in an upcoming stable release.

Reviewer: This work has now landed in Dart 2.19 so perhaps we should be explicit:

```suggestion
Support for this flag will be removed in Dart 2.19.
```

in immediate dependencies (`--precompile`).
To prevent precompilation, use `--no-precompile`.

### `--legacy-packages-file`

> If you are using Dart 2.18 and have any third-party tools
> which still use the discontinued `.packages` file,
> use `--legacy-packages-file` to generate a `.packages` file.
> Support for this flag will be removed in Dart 2.19.

Reviewer 2: ```suggestion
If you're using Dart 2.18 (or later) and use any third-party tools
that rely on the discontinued `.packages` file,
use `--legacy-packages-file` to generate a `.packages` file.
Support for this flag will be removed in Dart 2.19.
```

Reviewer: It seems the flag was removed in Dart 2.19 already, so do we need to add the "(or later)"?
